rules/library.bzl

```diff
@@ -980,19 +980,26 @@ def apple_library(
         #enable_framework_vfs = enable_framework_vfs
     )
 
-    # Generate resource bundles
-    module_data = library_tools["wrap_resources_in_filegroup"](
-        name = name + "_wrapped_resources_filegroup",
-        srcs = data,
-        testonly = testonly,
-    )
-    resource_bundles = library_tools["resource_bundle_generator"](
-        name = name,
-        library_tools = library_tools,
-        resource_bundles = kwargs.pop("resource_bundles", {}),
-        platforms = platforms,
-    )
-    deps += resource_bundles
+    # Wrap resources in a filegroup if requested.
+    if library_tools["wrap_resources_in_filegroup"]:
+        # Override the data as all the resources are now wrapped in a single filegroup target
+        data = [
+            library_tools["wrap_resources_in_filegroup"](
+                name = name + "_wrapped_resources_filegroup",
+                srcs = data,
+                testonly = testonly,
+            ),
+        ]
+
+    # Generate resource bundles for any requested resource files.
+    if library_tools["resource_bundle_generator"]:
```

Comment:
Is it necessary to set resource_bundle_generator to None? It's no-op if a target doesn't have `resource_bundles`.  If we want to make it explicit, we probably should raise if `resource_bundle_generator` is None && `resource_bundles` is not empty.

Reply:
I see what you mean though about this not being used if the resource_bundles attr isn't set (which is the case I care about) so I'll back this change out and keep just the wrap resources disabling ability